src/plugins/condition/criterion/TelemetryCriterion.js

```diff
@@ -72,17 +72,19 @@ export default class TelemetryCriterion extends EventEmitter {
     }
 
     computeResult(data) {
-        let comparator = this.findOperation(this.operation);
-        let params = [];
         let result = false;
-        params.push(data[this.metadata]);
-        if (this.input instanceof Array && this.input.length) {
-            params.push(this.input[0]);
-        } else if (this.input) {
-            params.push(this.input);
-        }
-        if (typeof comparator === 'function') {
-            result = comparator(params);
+        if (data) {
+            let comparator = this.findOperation(this.operation);
+            let params = [];
+            params.push(data[this.metadata]);
+            if (this.input instanceof Array && this.input.length) {
```

Comment:
This is the part we can simplify if we initialize output as an array. Can just check for length and do away with the else.

Reply:
Done